File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution 
{
    public double findMaxAverage(int[] nums, int k) 
    {
        int curr_sum=0,left=1,right=k;
        for(int i=0;i<k;i++)
        {
            curr_sum += nums[i];
        }
        int max_sum=curr_sum;
        for (int i = k; i < nums.length; i++) 
        {
            curr_sum += nums[i] - nums[i - k];
            max_sum = Math.max(max_sum, curr_sum);
        }
        return (double)max_sum/k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        curr_sum = sum(nums[:k])
        max_sum = curr_sum

        l = 1
        r = k

        while r < len(nums):
            curr_sum = curr_sum - nums[l-1] + nums[r]
            max_sum = max(max_sum, curr_sum)
            l += 1
            r += 1

        return max_sum / float(k)
